src/background/runtime.js

```diff
@@ -44,6 +44,9 @@ class Runtime {
         this.permissions.notifications = true;
       }
       this.storage.local.preferences = message.payload.preferences;
+      if(message.payload.migrate) {
+        this.migration.onUpdate({previousVersion: message.payload.previousVersion});
+      }
       await this.storage.persist();
```

Comment:
The migration should be called after the storage.persist.

Nit: whitespace after if

Reply:
Thanks!